src/api.ts

```diff
@@ -27,18 +28,19 @@ export function fetchJsonResponse<TValue, TError = string>(
  * Wraps a fetch request generated Promise in a ResultAsync,
  * ensuring both connection errors and Http status code errors
  * are converted into failed Results.
- * The response body is mapped to Unit. Use this for No Content responses.
+ * The raw Response object is returned as the Result success value.
  * @param request A Promise<Response> generated by a fetch() request
- * @param errorHandler Handles connection and Http status code errors
+ * @param errorHandler Handles connection and Http status code errors. Can be asynchronous to enable evaluation of the Response body.
  * @returns ResultAsync representing the success/failure of the request
  */
 export function fetchResponse<TError = string>(
   request: Promise<Response>,
-  errorHandler: (error: unknown) => Some<TError>
```

Comment:
Not sure why this wasn't retyped as `FunctionOfTtoK` originally 🤔. Must have missed it.